app/api/getAbsences/route.ts

```diff
@@ -0,0 +1,112 @@
+import { PrismaClient } from '@prisma/client';
+import { NextResponse } from 'next/server';
+
+let prisma: PrismaClient;
+
+if (process.env.NODE_ENV === 'production') {
+  prisma = new PrismaClient();
+} else {
+  if (!(global as any).prisma) {
+    (global as any).prisma = new PrismaClient();
+  }
+  prisma = (global as any).prisma;
+}
+
+export async function GET() {
+  try {
+    const absences = await prisma.absence.findMany({
+      select: {
+        id: true,
+        lessonDate: true,
+        subject: true,
+        lessonPlan: true,
+        reasonOfAbsence: true,
+        absentTeacher: {
+          select: {
+            id: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+          },
+        },
+        substituteTeacher: {
+          select: {
+            id: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+          },
+        },
+        location: {
+          select: {
+            id: true,
+            name: true,
+          },
+        },
+      },
+    });
+
+    if (!absences || absences.length === 0) {
+      return NextResponse.json({ events: [] }, { status: 200 });
+    }
+
+    const extractTimeArray = function (date: Date) {
+      return [
+        date.getFullYear(),
+        date.getMonth() + 1,
+        date.getDate(),
+        date.getHours(),
+        date.getMinutes(),
+      ];
+    };
+
+    const events = absences.map((absence) => {
+      const attendees = [
+        {
+          name: `${absence.absentTeacher.firstName} ${absence.absentTeacher.lastName}`,
+          email: absence.absentTeacher.email,
+          rsvp: true,
+          partstat: 'TENTATIVE',
+          role: 'OPT-PARTICIPANT',
+        },
+      ];
+
+      if (absence.substituteTeacher) {
+        attendees.push({
+          name: `${absence.substituteTeacher.firstName} ${absence.substituteTeacher.lastName}`,
+          email: absence.substituteTeacher.email,
+          rsvp: true,
+          partstat: 'TENTATIVE',
+          role: 'REQ-PARTICIPANT',
+        });
+      }
+
+      return {
+        start: extractTimeArray(absence.lessonDate),
+        duration: { hours: 2, minutes: 30 },
```

Comment:
Should the duration be 2 hours and 30 minutes? Shouldn't it be an all day event? I feel like only the day matters and it should be an all day event.

Reply:
Not sure how it is set up with sistema atm, but I am guessing that there are different start and end times in each event. I wanted to capture that, which is why I chose 2.5 hrs as a starting point. Can you check with the Yvonne on whether ICS needs to be full day or specific to a certain timeframe? Thanks

Reply:
The time during the day that the absence occurs doesn't matter. You are either absent for the entire day or are present. That's why the absences should be all day events

https://www.npmjs.com/package/ics
<img width="605" alt="image" src="https://github.com/user-attachments/assets/7ecb2138-0bff-4f95-939b-9f40cadaaf1c">

Reply:
Will do!